tldw_Server_API/app/api/v1/endpoints/mcp_unified_endpoint.py

```
@@ -496,10 +479,26 @@ async def get_server_metrics(
     return ServerMetricsResponse(**metrics)
 
 
+def _is_prometheus_public() -> bool:
+    """Flag indicating whether Prometheus metrics should be exposed without auth."""
+    return os.getenv("MCP_PROMETHEUS_PUBLIC", "").lower() in {"1", "true", "yes"}
+
+
+async def require_admin_unless_public(
+    public: bool = Depends(_is_prometheus_public),
+    user: TokenData = Depends(require_user),
```

Ruff (0.14.6)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)